testserver/binaries.go

```diff
@@ -124,19 +134,30 @@ func GetDownloadResponse(desiredVersion string, nonStable bool) (*http.Response,
 // To download the latest STABLE version of CRDB, set `nonStable` to false.
 // To download the bleeding edge version of CRDB, set `nonStable` to true.
 func DownloadBinary(tc *TestConfig, desiredVersion string, nonStable bool) (string, error) {
-	response, desiredVersion, err := GetDownloadResponse(desiredVersion, nonStable)
+	dbUrl, desiredVersion, err := GetDownloadURL(desiredVersion, nonStable)
 	if err != nil {
 		return "", err
 	}
 
-	defer func() { _ = response.Body.Close() }()
+	filename, err := GetDownloadFilename(desiredVersion)
+	if err != nil {
+		return "", err
+	}
+	localFile := filepath.Join(os.TempDir(), filename)
```

Comment:
curious how the short-circuiting will work. won't localFile be different each time this is called, since os.TempDir would return a different directory?

Reply:
No. You may be thinking of `MkdirTemp`.